### PR TITLE
Add `aws-lambda` as a feature to generated server SDK

### PR DESCRIPTION
**Pull request by drganjoo**
## Motivation and Context
The generated SDK conditionally re-exports `routing::LambdaHandler` under the feature flag aws-lambda, but the `Cargo.toml` does not define such a feature.

## Description
Closes: https://github.com/smithy-lang/smithy-rs/issues/3643

## Checklist
- [x] I have updated `CHANGELOG.next.toml` if I made changes to the smithy-rs codegen or runtime crates

### CODE REVIEW

File: CHANGELOG.next.toml [hunk 1]
```diff
@@ -65,3 +65,9 @@ message = "Update MSRV to `1.76.0`"
 references = ["smithy-rs#3653"]
 meta = { "breaking" = true, "tada" = true, "bug" = false, "target" = "all" }
 author = "landonxjames"
+
+[[smithy-rs]]
+message = "A feature, `aws-lambda`, is added to the generated SDK to re-export types required for Lambda deployment."
+references = ["smithy-rs#3643"]
+meta = { "breaking" = false, "bug" = true, "tada" = false }
```
Comment (david-perez) [inline CHANGELOG.next.toml:72]:
```suggestion
meta = { "breaking" = false, "bug" = true, "tada" = false, "target" = "server" }
```

File: CHANGELOG.next.toml [hunk 2]
```diff
@@ -65,3 +65,9 @@ message = "Update MSRV to `1.76.0`"
 references = ["smithy-rs#3653"]
 meta = { "breaking" = true, "tada" = true, "bug" = false, "target" = "all" }
 author = "landonxjames"
+
+[[smithy-rs]]
+message = "A feature, `aws-lambda`, is added to the generated SDK to re-export types required for Lambda deployment."
```
Comment (david-perez) [inline CHANGELOG.next.toml:70]:
```suggestion
message = "A feature, `aws-lambda`, has been added to generated SDKs to re-export types required for Lambda deployment."
```

File: examples/pokemon-service-lambda/src/main.rs [hunk 3]
```diff
@@ -5,14 +5,16 @@
...
     setup_tracing, stream_pokemon_radio, State,
 };
 use pokemon_service_lambda::get_storage_lambda;
-use pokemon_service_server_sdk::{PokemonService, PokemonServiceConfig};
+use pokemon_service_server_sdk::{
+    server::routing::LambdaHandler, PokemonService, PokemonServiceConfig,
+};
 
 #[tokio::main]
 pub async fn main() {
```
Comment (david-perez) [inline examples/pokemon-service-lambda/src/main.rs:20]: Shouldn't the `Cargo.toml` of this example project use the new feature and toggle it off from `aws-smithy-http-server`?

**Review by david-perez — APPROVED** [review]

File: examples/pokemon-service-lambda/src/main.rs (hunk 3 above)
Comment (drganjoo) [reply examples/pokemon-service-lambda/src/main.rs:20]: Yes, you have a point. This should have been a compilation error. Let me check why it compiles.

**Review by david-perez — APPROVED** [review]

**Pull request merged**